README.md

```diff
@@ -1,2 +1,138 @@
 # bolt11
 A library for encoding and decoding lightning network payment requests as defined in [BOLT #11](https://github.com/lightningnetwork/lightning-rfc/blob/master/11-payment-encoding.md).
+
+
+## Installation
+``` bash
+npm install bolt11
+```
+
+## Setup
+### Node.js
+``` javascript
+var lightningPayReq = require('bolt11')
+```
+
+
+## Examples
+### Decoding
+``` javascript
+var decoded = lightningPayReq.decode('lnbc20m1pvjluezhp58yjmdan79s6qqdhdzgynm4zwqd5d7xmw5fk98klysy043l2ahrqspp5qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqfppqw508d6qejxtdg4y5r3zarvary0c5xw7kepvrhrm9s57hejg0p662ur5j5cr03890fa7k2pypgttmh4897d3raaq85a293e9jpuqwl0rnfuwzam7yr8e690nd2ypcq9hlkdwdvycqa0qza8')
+/* decoded == below
+{
+  "coinType": "bitcoin",
+  "complete": true,
+  "satoshis": 2000000,
+  "payeeNodeKey": "03e7156ae33b0a208d0744199163177e909e80176e55d97a2f221ede0f934dd9ad",
+  "paymentRequest": "lnbc20m1pvjluezhp58yjmdan79s6qqdhdzgynm4zwqd5d7xmw5fk98klysy043l2ahrqspp5qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqfppqw508d6qejxtdg4y5r3zarvary0c5xw7kepvrhrm9s57hejg0p662ur5j5cr03890fa7k2pypgttmh4897d3raaq85a293e9jpuqwl0rnfuwzam7yr8e690nd2ypcq9hlkdwdvycqa0qza8",
+  "prefix": "lnbc20m",
+  "recoveryFlag": 0,
+  "signature": "c8583b8f65853d7cc90f0eb4ae0e92a606f89caf4f7d65048142d7bbd4e5f3623ef407a75458e4b20f00efbc734f1c2eefc419f3a2be6d51038016ffb35cd613",
+  "tags": [
+    {
+      "tagName": "purpose_commit_hash",
+      "data": "3925b6f67e2c340036ed12093dd44e0368df1b6ea26c53dbe4811f58fd5db8c1"
+    },
+    {
+      "tagName": "payment_hash",
+      "data": "0001020304050607080900010203040506070809000102030405060708090102"
+    },
+    {
+      "tagName": "fallback_address",
+      "data": {
+        "code": 0,
+        "address": "bc1qw508d6qejxtdg4y5r3zarvary0c5xw7kv8f3t4",
+        "addressHash": "751e76e8199196d454941c45d1b3a323f1433bd6"
+      }
+    }
+  ],
+  "timestamp": 1496314658,
+  "timestampString": "2017-06-01T10:57:38.000Z",
+  "wordsTemp": "temp1pvjluezhp58yjmdan79s6qqdhdzgynm4zwqd5d7xmw5fk98klysy043l2ahrqspp5qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqfppqw508d6qejxtdg4y5r3zarvary0c5xw7kepvrhrm9s57hejg0p662ur5j5cr03890fa7k2pypgttmh4897d3raaq85a293e9jpuqwl0rnfuwzam7yr8e690nd2ypcq9hlkdwdvycqawny4p"
+}
+*/
+```
+
+### Encoding
+* MINIMUM NEED: `privateKey` and one `payment_hash` tag as well as one `description`
```

Comment:
> MINIMUM NEED: `privateKey` and one `payment_hash` tag as well as one `description`

> `description` defaults to empty string

Confusing, these seem to be at odds with each other

Reply:
Good point.

Reply:
True.

What I mean by it is "the spec requires a signature, a payment hash and a description. The only one I can make an assumption for is description which we can default to an empty string."

Not sure if I should throw an error if description is missing... blegh...

Reply:
@junderw if the spec supports an empty string... support the default `''`

Reply:
@dcousens ok done! (that's how it already was)